apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/MutationsTest.cs

```diff
@@ -0,0 +1,106 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    public class MutationsTest
+    {
+        public static IEnumerable<object[]> ValidFamilyNames { get; } = new List<object[]>
+        {
+            new [] { "a" },
+            new [] { "A" },
+            new [] { "-_.aZ09" }
+        };
+
+        public static IEnumerable<object[]> InvalidFamilyNames { get; } = new List<object[]>
+        {
+            new [] { default(string) },
+            new [] { "" },
+            new [] { "abc*" },
+            new [] { "abc " },
+            new [] { "a=b" }
+        };
+
+        [Theory]
+        [MemberData(nameof(ValidFamilyNames))]
+        public void DeleteFromColumn(string familyName)
+        {
+            var mutation =
+                Mutations.DeleteFromColumn(familyName, "cq1", new BigtableVersionRange(1, 2));
+            Assert.NotNull(mutation.DeleteFromColumn);
+            Assert.Equal(familyName, mutation.DeleteFromColumn.FamilyName);
+            Assert.Equal("cq1", mutation.DeleteFromColumn.ColumnQualifier.ToStringUtf8());
+            Assert.Equal(1, mutation.DeleteFromColumn.TimeRange.StartTimestampMicros);
+            Assert.Equal(2, mutation.DeleteFromColumn.TimeRange.EndTimestampMicros);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidFamilyNames))]
+        public void DeleteFromColumnInvalidFamilyName(string familyName)
+        {
+            Assert.Throws<ArgumentException>(
+                () => Mutations.DeleteFromColumn(familyName, default(string), null));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidFamilyNames))]
+        public void DeleteFromFamily(string familyName)
+        {
+            var mutation = Mutations.DeleteFromFamily(familyName);
+            Assert.NotNull(mutation.DeleteFromFamily);
+            Assert.Equal(familyName, mutation.DeleteFromFamily.FamilyName);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidFamilyNames))]
+        public void DeleteFromFamilyInvalidFamilyName(string familyName)
+        {
+            Assert.Throws<ArgumentException>(() => Mutations.DeleteFromFamily(familyName));
+        }
+
+        [Fact]
+        public void DeleteFromRow()
+        {
+            var mutation = Mutations.DeleteFromRow();
+            Assert.NotNull(mutation.DeleteFromRow);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidFamilyNames))]
+        public void SetCell(string familyName)
+        {
+            var mutation =
+                Mutations.SetCell(familyName, "cq1", "value", new BigtableVersion(3));
```

Comment:
Are there other types you can set the cell value to?

Reply:
Yes, an arbitrary byte string and an 8-byte big-endian int64. Added tests for these as well.